cobalt/media/player/web_media_player_impl.cc

```diff
@@ -268,6 +268,9 @@ void WebMediaPlayerImpl::LoadMediaSource() {
       BIND_TO_RENDER_LOOP(
           &WebMediaPlayerImpl::OnEncryptedMediaInitDataEncounteredWrapper),
       media_log_));
+  if (is_video_buffer_budget_override_enabled_) {
```

Comment:
The implementation can be more simplified if we pass this to WebMediaPlayerImpl ctor, and allow the setting to be applied only to new playbacks after setting.  This will also resolve the threading issue in SourceBufferStream.

Reply:
Now passed into the SourceBufferStream constructor to avoid this condition.